client/landing/gutenboarding/hooks/use-podcast-title.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useState } from 'react';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { useAnchorFmParams } from '../path';
+
+export default function usePodcastTitle(): string | null {
+	const [ podcastTitle, setPodcastTitle ] = useState< string | null >( '' );
+	const { anchorFmPodcastId } = useAnchorFmParams();
+	interface PodcastDetailsObject {
+		title?: string;
+	}
+
+	useEffect( () => {
+		if ( ! anchorFmPodcastId ) {
+			return;
+		}
+		// Fetch podcast title from /podcast-details endpoint
+		( apiFetch( {
+			path: `https://public-api.wordpress.com/wpcom/v2/podcast-details?url=https://anchor.fm/s/${ encodeURIComponent(
+				anchorFmPodcastId
+			) }/podcast/rss&_fields=title`,
+		} ) as Promise< PodcastDetailsObject > )
```

Comment:
Rather than use an [overload](https://www.typescriptlang.org/docs/handbook/functions.html#overloads) (`as Promise< PodcastDetailsObject >`) which I think forces the type and overrides the type declaration, you can call the function with the [generic](https://www.typescriptlang.org/docs/handbook/generics.html#working-with-generic-type-variables) specified

```
apiFetch< PodcastDetailsObject >( ...
```

Reply:
Thanks! I made the change.